htlcswitch/link.go

```diff
@@ -2327,17 +2319,11 @@ func (l *channelLink) htlcSatifiesPolicyOutgoing(policy ForwardingPolicy,
 			"outgoing_expiry=%v, best_height=%v", payHash[:],
 			timeout, heightNow)
 
-		var failure lnwire.FailureMessage
-		update, err := l.cfg.FetchLastChannelUpdate(
-			l.ShortChanID(),
+		return l.createFailureWithUpdate(
+			func(upd *lnwire.ChannelUpdate) lnwire.FailureMessage {
+				return lnwire.NewExpiryTooSoon(*upd)
```

Comment:
not related to this pr, but i notice some of failure constructors take a pointer while others take a value. are some of them truly optional or it just an inconsistency in the API?

Reply:
I think it is correct. The only one that is optional that I saw is `FailTemporaryChannelFailure` and indeed in the decode we recognize a missing update.